package/MDAnalysis/coordinates/TRJ.py

```diff
@@ -640,7 +643,10 @@ def _read_frame(self, frame):
                 self.n_frames))
         # note: self.trjfile.variables['coordinates'].shape == (frames, n_atoms, 3)
         ts._pos[:] = self._get_var_and_scale('coordinates', frame)
-        ts.time = self._get_var_and_scale('time', frame)
+        try:
```

Comment:
to avoid hitting a try except at each read (which can be quite expensive) and keep to the same formatting as the other optional variables, could you instead just assign as `has_time` flag on line 550?

Reply:
I agree, done as you suggested!